akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
That is the case as soon as you add any stage to the flow. One could argue that Sink.publisher must always return a single-subscription publisher. We have even documented it as "can handle one [[org.reactivestreams.Subscriber]]".

I can turn the marker interface the other way around if Endre's point is considered a problem. I just thought it would be more places to mark.